debugtools/DDR_VM/src/com/ibm/j9ddr/view/dtfj/image/J9DDRImageProcess.java

```diff
@@ -419,34 +400,457 @@ public Iterator<?> getRuntimes() {
+		 * 22) SIGTTOU     23) SIGIO       24) SIGQUIT     25) SIGTSTP     26) SIGTRAP
+		 * 28) SIGWINCH    29) SIGXCPU     30) SIGXFSZ     31) SIGVTALRM   32) SIGPROF
+		 * 33) SIGDANGER
+		 */
+
+		switch (signal) {
+		case 1:
+			return "SIGHUP";
+		case 2:
+			return "SIGINT";
+		case 3:
+			return "SIGABRT";
+		case 4:
+			return "SIGILL";
+		case 5:
+			return "SIGPOLL";
+		case 6:
+			return "SIGURG";
+		case 7:
+			return "SIGSTOP";
+		case 8:
+			return "SIGFPE";
+		case 9:
+			return "SIGKILL"; // included for completeness, even though it can't be caught
+		case 10:
+			return "SIGBUS";
+		case 11:
+			return "SIGSEGV";
+		case 12:
+			return "SIGSYS";
+		case 13:
+			return "SIGPIPE";
+		case 14:
+			return "SIGALRM";
+		case 15:
+			return "SIGTERM";
+		case 16:
+			return "SIGUSR1";
+		case 17:
+			return "SIGUSR2";
+		case 19:
+			return "SIGCONT";
+		case 20:
+			return "SIGCHLD";
+		case 21:
+			return "SIGTTIN";
+		case 22:
+			return "SIGTTOU";
+		case 23:
+			return "SIGIO";
+		case 24:
+			return "SIGQUIT";
+		case 25:
+			return "SIGTSTP";
+		case 26:
+			return "SIGTRAP";
+		case 28:
+			return "SIGWINCH";
+		case 29:
+			return "SIGXCPU";
+		case 30:
+			return "SIGXFSZ";
+		case 31:
+			return "SIGVTALRM";
+		case 32:
+			return "SIGPROF";
+		case 33:
+			return "SIGDANGER";
+		default:
 			return null;
-		} else if (number > 0 && number < SIGNAL_NAMES.length) {
-			return SIGNAL_NAMES[number];
-		} else {
-			return "Unknown signal number " + number;
 		}
 	}
 
+	public String getSignalName() throws DataUnavailable, CorruptDataException {
+		String name = null;
+		int signal = getSignalNumber();
+
+		if (signal != 0) {
+			switch (process.getPlatform()) {
+			case AIX:
+				name = aixSignalName(signal);
+				break;
+			case LINUX:
+				name = linuxSignalName(signal);
+				break;
+			case OSX:
+				name = osxSignalName(signal);
+				break;
+			case WINDOWS:
+				name = windowsSignalName(signal);
+				break;
+			case ZOS:
+				name = zosSignalName(signal);
+				break;
+			default:
+				break;
+			}
+
+			if (name == null) {
+				name = "Unknown signal number " + signal;
+			}
+		}
+
+		return name;
+	}
+
 	public int getSignalNumber() throws DataUnavailable, CorruptDataException {
 		checkFailureInfo();
 
 		try {
-			if (j9rasProcessData.gpInfo() != null) {
-				int genericSignalNumber = (int) longByResolvingRawKey(j9rasProcessData.gpInfo(),
-						"J9Generic_Signal_Number");
-				if (genericSignalNumber == 0) {
-					// on Windows AMD64 the key name is J9Generic_Signal, so try that if we haven't got a signal number
-					genericSignalNumber = (int) longByResolvingRawKey(j9rasProcessData.gpInfo(), "J9Generic_Signal");
+			String gpInfo = j9rasProcessData.gpInfo();
+
+			if (gpInfo != null) {
+				// Names returned for OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE.
+				String[] keys;
+
+				switch (process.getPlatform()) {
+				case AIX:
+				case LINUX:
+				case OSX:
+					keys = new String[] { "Signal_Number" };
```

Comment:
Where do these names ("Signal_Number", "Windows_ExceptionCode", etc.) come from?

Reply:
Those names are defined by OMR in the various `omrsignal_context.c` files; e.g. [this one](https://github.com/eclipse/omr/blob/dd1373f8fbe46759cc02c3c6386dbee9ce30ce5e/port/linuxamd64/omrsignal_context.c#L51) for Linux-x64. Others are defined near uses of [`OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE`](https://github.com/eclipse/omr/search?q=OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE).

Reply:
nm. I see the comment now, I forgot to search OMR.